### PR TITLE
[IOS-607, IOS-611]: Profiles search on tiny delay and request cancelling

### PR DESCRIPTION
**Pull request by yuliaveres**
Make a search network request after 0.2 delay instead of immediately after each symbol is typed in;
cancel each previous request

### CODE REVIEW

File: Toshi/Controllers/UsersBotsGroups/SearchProfilesDataSource.swift [hunk 1]
```diff
@@ -54,19 +56,25 @@ final class SearchProfilesDataSource: NSObject {
         queryData.searchText = text
```
Comment (Elland) [inline Toshi/Controllers/UsersBotsGroups/SearchProfilesDataSource.swift]: Do we need the empty line above?

**Comment by yuliaveres** [comment]
@Elland  shit not everything is pushed 😭  adding

File: Toshi/Controllers/UsersBotsGroups/SearchProfilesDataSource.swift [hunk 2]
```diff
@@ -54,19 +56,25 @@ final class SearchProfilesDataSource: NSObject {
...
         queryData.type = type
 
-        idAPIClient.searchProfilesOfType(queryData.type, for: queryData.searchText) { [weak self] profiles, type, error in
+        reloadTimer?.invalidate()
```
Comment (designatednerd) [inline Toshi/Controllers/UsersBotsGroups/SearchProfilesDataSource.swift]: reload timer is invalidated here

**Comment by yuliaveres** [comment]
@designatednerd  any thoughts ?

**Pull request merged**